jolt-core/src/subprotocols/mles_product_sum.rs

```diff
@@ -1,19 +1,17 @@
 use std::iter::zip;
 
-use rayon::iter::{IndexedParallelIterator, IntoParallelRefIterator, ParallelIterator};
+use rayon::prelude::*;
 
 use crate::{
     field::{JoltField, MulU64WithCarry},
-    poly::{
-        eq_poly::EqPolynomial, multilinear_polynomial::MultilinearPolynomial, unipoly::UniPoly,
-    },
+    poly::{eq_poly::EqPolynomial, ra_poly::RaPolynomial, unipoly::UniPoly},
 };
 
 /// Computes the univariate polynomial `g(X) = sum_j eq((r', X, j), r) * prod_i mle_i(X, j)`.
 ///
 /// Note `claim` should equal `g(0) + g(1)`.
 pub fn compute_mles_product_sum<F: JoltField>(
-    mles: &[MultilinearPolynomial<F>],
+    mles: &[&RaPolynomial<F>],
```

Comment:
Can we have `&[RaPolynomial<F>]` here? To not do the iter().collect() or .each_ref() every time we call it.